cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -519,12 +567,20 @@ fn fastcall_prologue_epilogue(func: &mut ir::Function, isa: &dyn TargetIsa) -> C
     // will adjust the stack pointer to make room for the rest of the required
     // space for this frame.
     let word_size = isa.pointer_bytes() as usize;
-    let csr_stack_size = ((csrs.iter(GPR).len() + 2) * word_size) as i32;
+    let num_fprs = csrs.iter(FPR).len();
+    let mut csr_stack_size = ((csrs.iter(GPR).len() + 2) * word_size) as i32;
+
+    if num_fprs != 0 {
+        csr_stack_size += (num_fprs * types::F64X2.bytes() as usize) as i32;
+
+        // Ensure that csr stack space is 16-byte aligned for ideal SIMD value placement.
+        csr_stack_size = csr_stack_size & !0b1111 + 16;
```

Comment:
`csr_stack_size = (csr_stack_size + 15) & !0b1111;`

would save extending when already aligned to 16 bytes (unless I'm misunderstanding something and that's desirable)

Reply:
that'll teach me for trying to do math. Good catch!